From 0bce6a66f8221d18c186f0f192680ac3f48f0fe4 Mon Sep 17 00:00:00 2001
From: saranya <saranya.gopal@intel.com>
Date: Mon, 2 Apr 2018 16:57:32 +0530
Subject: [PATCH 08/28] Fix issue with power key long press

In Intel platforms like KBL, when power
key is pressed, both press and release events
are sent. So, this makes it impossible to detect
long press. We do not need suspend/resume feature
in KBL for HP nantucket project. So, just ignore
the power key release event.

Change-Id: I88045a9f6347fc459a825b371d505c48e8451499
Tracked-On:
Signed-off-by: saranya <saranya.gopal@intel.com>
---
 .../java/com/android/server/policy/PhoneWindowManager.java     | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/services/core/java/com/android/server/policy/PhoneWindowManager.java b/services/core/java/com/android/server/policy/PhoneWindowManager.java
index 28387c6e1fe0..23ee87851e14 100644
--- a/services/core/java/com/android/server/policy/PhoneWindowManager.java
+++ b/services/core/java/com/android/server/policy/PhoneWindowManager.java
@@ -1469,6 +1469,7 @@ public class PhoneWindowManager implements WindowManagerPolicy {
                 }
             }
         }
+        finishPowerKeyPress();
     }
 
     private void interceptPowerKeyUp(KeyEvent event, boolean interactive, boolean canceled) {
@@ -6428,8 +6429,6 @@ public class PhoneWindowManager implements WindowManagerPolicy {
                 }
                 if (down) {
                     interceptPowerKeyDown(event, interactive);
-                } else {
-                    interceptPowerKeyUp(event, interactive, canceled);
                 }
                 break;
             }
-- 
2.17.1

